src/app/rundown-view/services/mini-shelf-navigation.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core'
+import { Segment } from '../../core/models/segment'
+import { Rundown } from '../../core/models/rundown'
+import { Part } from '../../core/models/part'
+import { Tv2Part } from '../../core/models/tv2-part'
+import { Tv2Action, Tv2ActionContentType, Tv2VideoClipAction } from '../../shared/models/tv2-action'
+
+@Injectable()
+export class MiniShelfNavigationService {
+  public getNextMiniShelfSegment(rundown: Rundown, rundownActions: Tv2Action[]): Segment {
+    const currentActionId: string | undefined = this.getCurrentMiniShelfActionId(rundown)
+    const miniShelfSegmentGroup: Segment[] = this.getNearestMiniShelfSegmentGroupBeneathOnAirSegment(rundown)
+    const currentMiniShelfSegmentIndex: number = currentActionId ? this.getCurrentMiniShelfSegmentIndexFromActionId(currentActionId, miniShelfSegmentGroup, rundownActions) : -1
+
+    const nextMiniShelfSegmentIndex: number = (currentMiniShelfSegmentIndex + 1) % miniShelfSegmentGroup.length
+    return miniShelfSegmentGroup[nextMiniShelfSegmentIndex]
+  }
+
+  private getCurrentMiniShelfActionId(rundown: Rundown): string | undefined {
+    const onAirPart: Tv2Part = this.getOnAirPart(rundown)
+    const nextPart: Tv2Part = this.getNextPart(rundown)
+    return nextPart?.metadata?.actionId ?? onAirPart.metadata?.actionId
+  }
+
+  private getOnAirPart(rundown: Rundown): Tv2Part {
+    const nextPart: Part | undefined = rundown.segments.find(segment => segment.isOnAir)?.parts.find(part => part.isOnAir)
+
```

Comment:
This is just personal preference, but everywhere else i found we usually dont have the newline between the constant and the if-statement checking it.

Reply:
If you change it, remember that it is also the case for the `getNextPart` method.